R/kmeans.R

```diff
@@ -63,29 +64,25 @@ exp_kmeans <- function(df, ...,
                                                 seed=NULL) # Seed is already done. Skip it.
   }
 
-  # TODO: Running PCA is not necessary for elbow method case.
-  ret <- do_prcomp(df, normalize_data = normalize_data, seed = NULL, ...)
-
   if (!elbow_method_mode) {
+    ret <- do_prcomp(df, normalize_data = normalize_data, seed = NULL, ...)
     ret <- ret %>% dplyr::mutate(model = purrr::map2(model, !!kmeans_model_df$model, function(x, y) {
       x$kmeans <- y # Might need to be more careful on guaranteeing x and y are from same group, but we are not supporting group_by on UI at this point.
       x$sampled_nrow <- sampled_nrow
       x
     }))
   }
   else {
-    ret <- ret %>% ungroup()
-    ret <- ret %>% dplyr::mutate(model = purrr::map(model, function(x) {
-      kmeans_df <- as.data.frame(x$x)
-      ret <- iterate_kmeans(kmeans_df,
-                            max_centers = max_centers,
-                            iter.max = iter.max,
-                            nstart = nstart,
-                            algorithm = algorithm,
-                            trace = trace,
-                            seed=NULL) # Seed is already done in do_prcomp. Skip it.
-      ret
-    }))
+    kmeans_df <- df %>% dplyr::select(!!!rlang::quos(...))
+    ret <- iterate_kmeans(kmeans_df,
+                          max_centers = max_centers,
+                          iter.max = iter.max,
+                          nstart = nstart,
+                          algorithm = algorithm,
+                          trace = trace,
+                          normalize_data = normalize_data,
+                          seed=NULL) # Seed is already done in do_prcomp. Skip it.
+    ret <- data.frame(model = I(list(ret))) # Follow current output format for now. TODO: Revisit and support group_by.
```

Comment:
```suggestion
    ret <- data.frame(model = I(list(ret))) # Follow current output format for now. I() is to avoid unwanted expansion of list at creation of data frame. TODO: Revisit and support group_by.
```